Core/Business/Checkout/TermsAndConditions.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Business\Checkout;
+
+class TermsAndConditions
+{
+    private $identifier;
+    private $links;
+    private $rawText;
+
+    public function setText($rawText, ...$links)
+    {
+        $this->links = $links;
+        $this->rawText = $rawText;
+        return $this;
+    }
+
+    /**
+     * Inserts links into the text, replacing all [something] with links to "something", taking
+     * URLs from $this->links
+     * @return an string of HTML
+     */
+    public function format()
+    {
+        $index = 0;
+        return preg_replace_callback('/\[(.*?)\]/', function (array $match) use (&$index) {
+            if (!isset($this->links[$index])) {
+                return $match[1];
+            }
+
+            $replacement = '<a target="_blank" href="' . $this->links[$index] . '">' . $match[1] . '</a>';
```

Comment:
nah, I think `data-target` will be better these days and global js `$('[data-target=external]').prop('target', '_blank');` (or by query selector without jquery) just to keep it w3c validated